src/tennisSimulator/match/Set.java

```diff
@@ -0,0 +1,111 @@
+package tennisSimulator.match;
+
+import tennisSimulator.config.Configuration;
+import tennisSimulator.player.Player;
+
+public class Set {
+
+    private final Player firstPlayer;
+    private final Player secondPlayer;
+    private final ScoreTable scoreTable;
+    private final Game game;
+    private final int games;
+    private Integer firstPlayerGamesWon;
+    private Integer secondPlayerGamesWon;
+
+    public Set(Player firstPlayer, Player secondPlayer, ScoreTable scoreTable) {
+        this.firstPlayer = firstPlayer;
+        this.secondPlayer = secondPlayer;
+        this.scoreTable = scoreTable;
+        game = new Game(firstPlayer, secondPlayer, scoreTable);
+        this.games = Configuration.games;
+        firstPlayerGamesWon = 0;
+        secondPlayerGamesWon = 0;
+    }
+
+    public void simulate() {
+        while (hasNoWinner()) {
+            Player gameWinner = game.simulate();
+            if (gameWinner.equals(firstPlayer)) {
```

Comment:
No purpose of having if/else statement here. Simply call updateResult(gameWinner);

Reply:
fixed